mdms/mdms-analytics/src/main/scala/de/hpi/isg/mdms/analytics/TableImportance.scala

```diff
@@ -0,0 +1,144 @@
+    * @param metadataStore  is the metadata store
+    * @param idUtils utility tools
+    * @param columnStatistics is constraint collection column statistics
+    * @param epsilon if Euclidean Distance between V and V+1  is
+    *                lower epsilon, the stationary distribution is reached
+    * @param numIteration is the maximum number of iteration
+    * @param maxnumOrEpsilon true: solution is found if maximum number of iteration is reached
+    *                        false: solution is found if changes of solution vector is below a critical threshold
+    * @return the solution vector containing the table importance
+    */
+  def tableImport(probMatrix: DataQuanta[(Int, Int, Double)],
+                  tupleCount: ConstraintCollection[TupleCount],
+                  metadataStore: MetadataStore,
+                  idUtils: IdUtils,
+                  columnStatistics: ConstraintCollection[ColumnStatistics],
+                  epsilon: Double = 1e-10,
+                  numIteration: Int = 100,
+                  maxnumOrEpsilon: Boolean = true): DataQuanta[(Int, Double)] = {
+    val Vinitial = initiateVector(tupleCount, metadataStore, idUtils, columnStatistics)
+    // Looping until reaching chosen max number of iteration
+    if (maxnumOrEpsilon) Vinitial.repeat(numIteration, Vold => iteratingImportance(probMatrix, Vold))
+    // Looping until changes of vectors is below a critical threshold
+    else Vinitial.doWhile[Double](_.head < epsilon, { Vold =>
+        val Vnew = iteratingImportance(probMatrix, Vold)
+        val Vdiff = diffV(Vnew, Vold).map(a => math.sqrt(a))
+        (Vnew, Vdiff.filter { x => println(x); true })
+      })
+  }
+
+  // Calculating the the product of the probability vector and the solution vector
+  def iteratingImportance(probMatrix: DataQuanta[(Int, Int, Double)],
+                          V: DataQuanta[(Int, Double)]): DataQuanta[(Int, Double)] = {
+    return probMatrix
+      .keyBy(a => a._1).keyJoin(V.keyBy(_._1)).assemble((a, b) => (a._1, a._2, a._3 * b._2))
+      .reduceByKey((_._2), (a, b) => (a._1, a._2, a._3 + b._3))
+      .map(a => (a._2, a._3))
+  }
+
+  // Calculating the Euclidean Distance between the solution vector V and V+1
+  def diffV(Vnew: DataQuanta[(Int, Double)], Vold: DataQuanta[(Int, Double)]): DataQuanta[(Double)] = {
+    val diffVnewVold = Vnew.keyBy(_._1).keyJoin(Vold.keyBy(_._1)).assemble((a, b) => (a._2, b._2))
+      .map(a => ((a._1 - a._2) * (a._1 - a._2))).reduce(_ + _)
+    return diffVnewVold
+  }
+
+  // Initiating the solution vector
+  def initiateVector(tupleCount: ConstraintCollection[TupleCount],
+                     metadataStore: MetadataStore,
+                     idUtils: IdUtils,
+                     columnStatistics: ConstraintCollection[ColumnStatistics]):DataQuanta[(Int, Double)] = {
+    // R ... number of tuples of table
+    val R = metadataStore.loadConstraints(tupleCount)
```

Comment:
Please don't start variable identifiers with capital letters. But it's okay here.